### PR TITLE
Add SQL editor to DB connector

### PR DESCRIPTION
**Pull request by chriddyp**
all yours @jackparmer :baseball:

### CODE REVIEW

**Comment by jackparmer** [comment]
This PR combines:

- Upgrade to Webpack 2 by @chriddyp . This was needed to speed up `yarn install` after adding `plotly.js`
-  An interface for the "Preview" tab to compose and run queries:
    * `react-codemirror` with autocomplete
    * A lightweight plotly.js chart editor
    * A SQL schema tree of tables and columns
    * (screenshot at the bottom)

This currently works for these connectors:
- [x] MySQL
- [x] Postgres
- [x] MsSql
- [x] MariaDB
- [x] Redshift
- [x] S3
- [x] ApacheDrill
- [x] SQlite
- [x ] SparkSQL

It's currently broken for these connectors:
- [ ] ElasticSearch
- [ ] DB2

@chriddyp this is probably ready for an initial review when you have time. I could also use your help in debugging the ES connection (copied a screenshot of the ES connection and errro below).

@tarzzz @n-riesco The Livy/Hive example connection works OK, but takes about a minute to connect. It also seems that only one connection at a time is allowed. I guess this is because, "the current server is unable of creating a HiveContext if another session has already created one" as @n-riesco mentioned in Slack. Are there ways around this? 
I had to comment out the IBM DB2 connector (https://github.com/plotly/plotly-database-connector/pull/178/commits/50df8990edb3815a0bcae565a04016b5974cd182) because I would get a `Could not locate the bindings file` error when starting Electron. Any tips for IBM DB2? Here's the full error:

```
App threw an error during load
Error: Could not locate the bindings file. Tried:
 → /Users/jackparmer/Documents/plotly/falcon/plotly-database-connector/node_modules/ibm_db/build/odbc_bindings.node
 → /Users/jackparmer/Documents/plotly/falcon/plotly-database-connector/node_modules/ibm_db/build/Debug/odbc_bindings.node
 → /Users/jackparmer/Documents/plotly/falcon/plotly-database-connector/node_modules/ibm_db/build/Release/odbc_bindings.node
 → /Users/jackparmer/Documents/plotly/falcon/plotly-database-connector/node_modules/ibm_db/out/Debug/odbc_bindings.node
 → /Users/jackparmer/Documents/plotly/falcon/plotly-database-connector/node_modules/ibm_db/Debug/odbc_bindings.node
 → /Users/jackparmer/Documents/plotly/falcon/plotly-database-connector/node_modules/ibm_db/out/Release/odbc_bindings.node
 → /Users/jackparmer/Documents/plotly/falcon/plotly-database-connector/node_modules/ibm_db/Release/odbc_bindings.node
 → /Users/jackparmer/Documents/plotly/falcon/plotly-database-connector/node_modules/ibm_db/build/default/odbc_bindings.node
 → /Users/jackparmer/Documents/plotly/falcon/plotly-database-connector/node_modules/ibm_db/compiled/7.9.0/darwin/x64/odbc_bindings.node
    at bindings (/Users/jackparmer/Documents/plotly/falcon/plotly-database-connector/node_modules/bindings/bindings.js:88:9)
    at Object.<anonymous> (/Users/jackparmer/Documents/plotly/falcon/plotly-database-connector/node_modules/ibm_db/lib/odbc.js:27:31)
    at Object.<anonymous> (/Users/jackparmer/Documents/plotly/falcon/plotly-database-connector/node_modules/ibm_db/lib/odbc.js:1770:3)
    at Module._compile (module.js:571:32)
    at Object.Module._extensions..js (module.js:580:10)
    at Module.load (module.js:488:32)
    at tryModuleLoad (module.js:447:12)
    at Function.Module._load (module.js:439:3)
    at Module.require (module.js:498:17)
    at require (internal/module.js:20:19)
    at Object.<anonymous> (/Users/jackparmer/Documents/plotly/falcon/plotly-database-connector/backend/main.js:3:35502)
    at t (/Users/jackparmer/Documents/plotly/falcon/plotly-database-connector/backend/main.js:2:101)
    at Object.defineProperty.value (/Users/jackparmer/Documents/plotly/falcon/plotly-database-connector/backend/main.js:2:1950833)
    at t (/Users/jackparmer/Documents/plotly/falcon/plotly-database-connector/backend/main.js:2:101)
    at Object.e.exports (/Users/jackparmer/Documents/plotly/falcon/plotly-database-connector/backend/main.js:2:94789)
    at t (/Users/jackparmer/Documents/plotly/falcon/plotly-database-connector/backend/main.js:2:101)
```


## elasticsearch

image

Seems to be an issue with the initial POST that gets the mappings to preview:
image

image

## Screenshot of the new preview tab

image

**Comment by jackparmer** [comment]
Got DB2 working with the SQL editor in the last commit 🎉  🌮  

image

File: package.json [hunk 1]
```diff
@@ -31,7 +31,7 @@
     "package-all": "yarn run package -- --all",
     "rebuild-and-e2e": "yarn run build && yarn run test-e2e",
     "start": "cross-env NODE_ENV=production electron ./",
-    "start-headless": "cross-env NODE_ENV=production electron ./dist/headless-bundle.js",
+    "start-headless": "cross-env NODE_ENV=production node ./dist/headless-bundle.js",
```
Comment (n-riesco) [inline package.json]: This is going to cause a lot of pain, because it'll require to rebuild native modules against node.

File: package.json [hunk 2]
```diff
@@ -167,19 +167,19 @@
...
     "electron-rebuild": "^1.1.3",
     "electron-sudo": "^3.0.13",
     "eslint-plugin-import": "^1.8.0",
-    "extract-text-webpack-plugin": "2",
+    "extract-text-webpack-plugin": "2.1.2",
     "fbjs-scripts": "^0.6.0",
     "font-awesome": "^4.6.1",
     "fs-extra": "https://registry.npmjs.org/fs-extra/-/fs-extra-0.26.7.tgz",
-    "ibm_db": "2.0.0",
+    "ibm_db": "^2.1.0",
```
Comment (n-riesco) [inline package.json]: ibm_db@2.1 doesn't work in electron.

**Comment by jackparmer** [comment]
Alright, finally got this working for all of the database connectors 🎉 

@n-riesco can you take it for a spin when you have a chance?

@alexcjohnson you might be interested in this as well, or could wait until it built as a release (hopefully next week). 

image

Here's how it looks for some non-SQL connectors:

### S3

image

### Elasticsearch

image

### An error message

image

**Comment by n-riesco** [comment]
I'll update the list of problems here as I find them:
- `yarn test` fails because of missing `webpack.config.node.js`

**Pull request merged**